pkg/polaris/polarisDBMutator.go

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (C) 2019 Synopsys, Inc.
+ *
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ *  with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package polaris
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	b64 "encoding/base64"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func GetPolarisDBComponents(baseUrl string, polaris Polaris) map[string]runtime.Object {
+	content, err := GetBaseYaml(baseUrl, "polaris", polaris.Version, "polarisdb_base.yaml")
+	if err != nil {
+		panic(err)
+	}
+
+	// regex patching
+	content = strings.ReplaceAll(content, "${NAMESPACE}", polaris.Namespace)
+	content = strings.ReplaceAll(content, "${ENVIRONMENT_NAME}", polaris.EnvironmentName)
+	content = strings.ReplaceAll(content, "${IMAGE_PULL_SECRETS}", polaris.ImagePullSecrets)
+	content = strings.ReplaceAll(content, "${POSTGRES_USERNAME}", polaris.PolarisDBSpec.PostgresDetails.Username)
+	content = strings.ReplaceAll(content, "${POSTGRES_PASSWORD}", polaris.PolarisDBSpec.PostgresDetails.Password)
+	content = strings.ReplaceAll(content, "${SMTP_HOST}", polaris.PolarisDBSpec.SMTPDetails.Host)
+	if polaris.PolarisDBSpec.SMTPDetails.Port != 2525 {
+		content = strings.ReplaceAll(content, "2525", strconv.Itoa(polaris.PolarisDBSpec.SMTPDetails.Port))
+	}
+	if len(polaris.PolarisDBSpec.SMTPDetails.Username) != 0 {
+		content = strings.ReplaceAll(content, "${SMTP_USERNAME}", EncodeStringToBase64(polaris.PolarisDBSpec.SMTPDetails.Username))
+	} else {
+		content = strings.ReplaceAll(content, "${SMTP_USERNAME}", "Cg==")
+	}
+	if len(polaris.PolarisDBSpec.SMTPDetails.Password) != 0 {
+		content = strings.ReplaceAll(content, "${SMTP_PASSWORD}", fmt.Sprintf("\"%s\"", EncodeStringToBase64(polaris.PolarisDBSpec.SMTPDetails.Password)))
+	} else {
+		content = strings.ReplaceAll(content, "${SMTP_PASSWORD}", "Cg==")
+	}
+	content = strings.ReplaceAll(content, "${POSTGRES_HOST}", polaris.PolarisDBSpec.PostgresDetails.Host)
+	if polaris.PolarisDBSpec.PostgresDetails.Port != 5432 {
```

Comment:
why this conversion?

Reply:
strings.ReplaceAll takes a string as parameter and polaris.PolarisDBSpec.PostgresDetails.Port is an integer.